### PR TITLE
feat(openchallenges): migrate the OC Angular app to Vite (ARCH-337)

### PR DESCRIPTION
**Pull request by tschaffter**
Closes https://sagebionetworks.jira.com/browse/ARCH-337

## Changelog

- Generate a new Angular app project configured to use Vite, update it and then replace the existing app project.
- Specify `stylePreprocessorOptions` in `project.json` to reduce the location where the bundle can find styles.
- Remove the sub-folder `lib` from the project `openchallenges-styles` and `openchallenges-themes`.
- Update the paths to styles and themes files after setting `stylePreprocessorOptions` and refactoring the styles and themes projects.
- Adapt the Dockerfile to the Vite build output.
- Remove custom code previously added to `server.ts` to inject the server port when using the Express server.
- Remove Tailwind CSS from the project `sandbox-angular-app`.
- Adapt `model-ad-app:build` to changes made to the shared styles.

### CODE REVIEW

File: apps/openchallenges/app/.eslintrc.json [hunk 1]
```diff
@@ -10,15 +10,15 @@
           "error",
           {
             "type": "attribute",
-            "prefix": "openchallenges",
```
Comment (tschaffter) [inline apps/openchallenges/app/.eslintrc.json:13]: #Reverting to the default value used by the Angular code generator for the sake of simplification for the next time we need to regenerate the app. This means that less changes will need to be make to the `index.html` and `app.component.ts` too.

File: apps/openchallenges/app/Dockerfile [hunk 2]
```diff
@@ -1,6 +1,6 @@
 FROM node:20.7.0-alpine
 
-ENV APP_DIR=/app
```
Comment (tschaffter) [inline apps/openchallenges/app/Dockerfile:3]: This new path is more standard based on my experience.

File: apps/openchallenges/app/src/_app-theme.scss [hunk 3]
```diff
@@ -1,15 +1,13 @@
 @use 'sass:map';
 @use '@angular/material' as mat;
-@use 'libs/openchallenges/themes/src/fonts' as fonts;
```
Comment (tschaffter) [inline apps/openchallenges/app/src/_app-theme.scss:3]: Note that the path does no longer include the prefix `libs/` thanks to specifying `stylePreprocessorOptions` in `project.json`.

File: apps/openchallenges/app/src/_app-theme.scss [hunk 4]
```diff
@@ -1,15 +1,13 @@
...
-@use 'libs/openchallenges/themes/src/palettes' as palettes;
-@use 'libs/openchallenges/themes/src/index' as openchallenges;
-
+@use 'openchallenges/themes/src/fonts' as fonts;
+@use 'openchallenges/themes/src/palettes' as palettes;
+@use 'openchallenges/themes/src/index' as openchallenges;
 @include mat.typography-hierarchy(fonts.$lato);
-@include mat.core();
```
Comment (tschaffter) [inline apps/openchallenges/app/src/_app-theme.scss:8]: I didn't make this change so I'm assuming it's the work of the auto-formatter.

File: apps/openchallenges/app/src/app/app.component.spec.ts.off
Comment (tschaffter) [inline apps/openchallenges/app/src/app/app.component.spec.ts.off:1]: I did not migrate our changes to this file after running the code generator.

File: tsconfig.base.json [hunk 5]
```diff
@@ -59,14 +58,13 @@
       "@sagebionetworks/openchallenges/org-search": ["libs/openchallenges/org-search/src/index.ts"],
       "@sagebionetworks/openchallenges/pages": ["libs/openchallenges/pages/src/index.ts"],
       "@sagebionetworks/openchallenges/signup": ["libs/openchallenges/signup/src/index.ts"],
-      "@sagebionetworks/openchallenges/styles": ["libs/openchallenges/styles/src/index.ts"],
```
Comment (tschaffter) [inline tsconfig.base.json:62]: Remove the styles and themes projects since they are not TS projects.

**Review by rrchai — APPROVED** [review]
LGTM and tested!

**Pull request merged**